packages/core/admin/utils/plugins.js

```diff
@@ -0,0 +1,218 @@
+ * @returns {Plugin[]}
+ */
+const getPlugins = (pluginsAllowlist) => {
+  const rootPath = path.resolve(__dirname, '..', path.join('..', '..', '..', 'packages'));
+  /**
+   * So `glob` only supports '/' as a path separator, so we need to replace
+   * the path separator for the current OS with '/'. e.g. on windows it's `\`.
+   *
+   * see https://github.com/isaacs/node-glob/#windows for more information
+   *
+   * and see https://github.com/isaacs/node-glob/issues/467#issuecomment-1114240501 for the recommended fix.
+   */
+  let corePath = path.join(rootPath, 'core', '*');
+  let pluginsPath = path.join(rootPath, 'plugins', '*');
+
+  if (process.platform === 'win32') {
+    corePath = corePath.split(path.sep).join(path.posix.sep);
+    pluginsPath = pluginsPath.split(path.sep).join(path.posix.sep);
+  }
+
+  const corePackageDirs = glob.sync(corePath);
+  const pluginsPackageDirs = glob.sync(pluginsPath);
+
+  const plugins = [...corePackageDirs, ...pluginsPackageDirs]
+    .map((directory) => {
+      const isCoreAdmin = directory.includes('packages/core/admin');
+
+      if (isCoreAdmin) {
+        return null;
+      }
+
+      const { name, strapi } = require(path.join(directory, 'package.json'));
+
+      /**
+       * this will remove any of our packages that are
+       * not actually plugins for the application
+       */
+      if (!strapi) {
+        return null;
+      }
+
+      /**
+       * we want the name of the node_module
+       */
+      return {
+        pathToPlugin: name,
+        name: strapi.name,
+        info: { ...strapi, packageName: name },
+        directory,
+      };
+    })
+    .filter(filterPluginsByAdminEntry);
+
+  if (Array.isArray(pluginsAllowlist)) {
+    return plugins.filter((plugin) => pluginsAllowlist.includes(plugin.pathToPlugin));
+  }
+
+  return plugins;
+};
+
+/**
+ * @type {(plugin: Plugin) => boolean}
+ */
+const filterPluginsByAdminEntry = (plugin) => {
+  if (!plugin) {
+    return false;
+  }
+
+  /**
+   * There are two ways a plugin should be imported, either it's local to the strapi app,
+   * or it's an actual npm module that's installed and resolved via node_modules.
+   *
+   * We first check if the plugin is local to the strapi app, using a regular `resolve` because
+   * the pathToPlugin will be relative i.e. `/Users/my-name/strapi-app/src/plugins/my-plugin`.
+   *
+   * If the file doesn't exist well then it's probably a node_module, so instead we use `require.resolve`
+   * which will resolve the path to the module in node_modules. If it fails with the specific code `MODULE_NOT_FOUND`
+   * then it doesn't have an admin part to the package.
+   *
+   * NOTE: we should try to move to `./package.json[exports]` map with bundling of our own plugins,
+   * because these entry files are written in commonjs restricting features e.g. tree-shaking.
+   */
+  try {
+    const isLocalPluginWithLegacyAdminFile = fs.existsSync(
+      path.resolve(`${plugin.pathToPlugin}/strapi-admin.js`)
```

Comment:
No, because `strapi-admin` was always at the root of the project and we wouldn't be able to read these because of typescript so this is what it was like prior to this work 👍🏼